src/details/ArborX_DetailsBatchedQueries.hpp

```diff
@@ -82,18 +84,30 @@ struct BatchedQueries
     return sortObjects(morton_codes);
   }
 
-  template <typename T>
-  static Kokkos::View<T *, DeviceType>
-  applyPermutation(Kokkos::View<size_t const *, DeviceType> permute,
-                   Kokkos::View<T *, DeviceType> v)
+  // NOTE  trailing return type seems required :(
+  // error: The enclosing parent function ("applyPermutation") for an extended
```

Comment:
Is this an NVCC limitation? I would assume so.

Reply:
Yep.